### PR TITLE
Feature/ar 561 quote template 06 2021

### PR DESCRIPTION
**Pull request by sinejespersen**
https://jira.itkdev.dk/browse/AR-561

It

* has transition
* changes every "duration" second
* has icon component
* the choice between 1 and 2 lines

It lacks

* theming (color/font)
* fontsize - as I am not sure about this
* the position is in the top - i am not sure how this should be implemented with the different regions - but can easily be changed

Picture time: 

Screenshot 2021-06-29 at 13 19 57

### CODE REVIEW

File: src/templates/quote/icon.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * @param {number} props.strokeWidth
+ *   The stroke width.
+ * @param {string} props.className
+ *   The class name.
+ * @returns {JSX.Element}
+ *   The component.
+ */
+function Icon({ stroke, strokeWidth, className }) {
+  return (
+    <svg xmlns="http://www.w3.org/2000/svg" width="38" height="28" className={className} viewBox="0 0 38 28">
```
Comment (cableman) [inline src/templates/quote/icon.js:18]: I have seen examples where svg is in svg files an using import to get content into an const. Would that not make editing these svg's later on easier?

File: src/templates/quote/icon.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import { React } from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * @param {object} props
```
Comment (cableman) [inline src/templates/quote/icon.js:5]: Would really like a function comment?

Comment (cableman) [inline src/templates/quote/icon.js:5]: I see no props parameter? Only see destruction of propes.

File: src/templates/quote/quote.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  const [first] = quotes;
+  const [currentQuote, setCurrentQuote] = useState(first);
+  const [show, setShow] = useState(true);
+
+  /**
+   * Setup slide run function.
+   */
+  const slideExecution = new BaseSlideExecution(slide, slideDone);
+  useEffect(() => {
+    if (run) {
```
Comment (cableman) [inline src/templates/quote/quote.js:35]: I am sure that i'm missing something... but where is `run` defined?

File: src/templates/quote/quote.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        clearInterval(animationTimer);
+      }
+    };
+  }, [currentQuote]);
+
+  return (
+    <>
+      <div className={show ? 'template-quote show' : 'template-quote hide'}>
+        {/* todo make this themeable */}
+        <Icon stroke="#EE0043" strokeWidth={2} className="quotation-mark" />
```
Comment (cableman) [inline src/templates/quote/quote.js:68]: Hardcode hex value for stroke?

**Review by cableman — CHANGES_REQUESTED** [review]

File: src/templates/quote/quote.js (hunk 4 above)
Comment (sinejespersen) [reply src/templates/quote/quote.js:68]: notice above: {/* todo make this themeable */} , i've inserted a hex value _for now_

File: src/templates/quote/quote.js (hunk 3 above)
Comment (sinejespersen) [reply src/templates/quote/quote.js:35]: above 

`function Quote({ slide, content, run, slideDone }) {`

File: src/templates/quote/icon.js (hunk 2 above)
Comment (sinejespersen) [reply src/templates/quote/icon.js:5]: The linter yells at me if I dont do it

**Review by turegjorup — APPROVED** [review]

**Review by cableman — APPROVED** [review]

**Pull request merged**